buildpacks/nodejs-yarn/src/layers/deps.rs

```diff
@@ -0,0 +1,100 @@
+use crate::yarn::Yarn;
+use crate::{YarnBuildpack, YarnBuildpackError};
+use libcnb::build::BuildContext;
+use libcnb::data::layer_content_metadata::LayerTypes;
+use libcnb::layer::{ExistingLayerStrategy, Layer, LayerData, LayerResult, LayerResultBuilder};
+use libcnb::Buildpack;
+use libherokubuildpack::log::log_info;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::Path;
+use thiserror::Error;
+
+/// `DepsLayer` is a layer for caching yarn dependencies from build to build.
+/// This layer is irrelevant in zero-install mode, as cached dependencies are
+/// included in the source code. Yarn only prunes unused dependencies in a few
+/// scenarios, so the cache is invalidated after a number of builds to prevent
+/// unbound cache growth.
+pub(crate) struct DepsLayer {
+    pub(crate) yarn: Yarn,
+}
+
+#[derive(Deserialize, Serialize, Clone, PartialEq)]
+pub(crate) struct DepsLayerMetadata {
+    // Using float here due to [an issue with lifecycle's handling of integers](https://github.com/buildpacks/lifecycle/issues/884)
+    cache_usage_count: f32,
+    layer_version: String,
```

Comment:
Seems like a good idea to futureproof metadata. I wonder if we should do more to promote this somehow (in the future).

Reply:
Yeah, I've done this on three buildpacks now. My thinking is that anytime I know I'm going to make a breaking change to the cache or metadata, I can bump this.